packages/web-components/src/components/va-breadcrumbs/va-breadcrumbs.tsx

```diff
@@ -0,0 +1,126 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  h,
+  Prop,
+  State,
+} from '@stencil/core';
+import { getSlottedNodes } from '../../utils/utils';
+
+@Component({
+  tag: 'va-breadcrumbs',
+  styleUrl: 'va-breadcrumbs.css',
+  shadow: true,
+})
+export class VaBreadcrumbs {
+  @Element() el: HTMLElement;
+
+  /**
+   * Adds an aria-label attribute to the <nav /> element.
+   */
+  @Prop() label: string = 'Breadcrumb';
+
+  /**
+   * Analytics tracking function(s) will not be called
+   */
+  @Prop() disableAnalytics: boolean;
+
+  /**
+   * Adds a custom id attribute to the UL element
+   */
+  @Prop() listId: string;
+
+  /**
+   * Adds CSS class `.va-nav-breadcrumbs--mobile` to the
+   * NAV element. The mobile breadcrumb will always
+   * be displayed while mobileFirstProp is True.
+   */
+  @Prop() mobileFirstProp: boolean = false;
+
+  /**
+   * Adds a custom id attribute to the NAV element
+   */
+  @Prop() navId: string;
+
+  @State() breadcrumbs: Array<Node>;
+
+  @Event({
+    eventName: 'component-library-analytics',
+    composed: true,
+    bubbles: true,
+  })
+  componentLibraryAnalytics: EventEmitter;
+
+  private fireAnalyticsEvent(event: MouseEvent): void {
+    if (!this.disableAnalytics) {
+      const target = event.target as HTMLElement;
+      // If it's a link being clicked, dispatch an analytics event
+      if (target?.tagName === 'A') {
+        const details = {
+          componentName: 'va-breadcrumbs',
+          action: 'linkClick',
+          details: {
+            clickLabel: target.innerText.trim(),
+            clickLevel: parseInt(target.dataset.index, 10) + 1,
+            totalLevels: getSlottedNodes(this.el, 'a').length,
+            mobileFirstProp: this.mobileFirstProp,
+          },
+        };
+        this.componentLibraryAnalytics.emit(details);
+      }
+    }
+  }
+
+  componentDidLoad() {
```

Comment:
When I ran `yarn test` locally I was receiving the following console warning
<img width="685" alt="Screen Shot 2021-12-10 at 2 26 22 PM" src="https://user-images.githubusercontent.com/11822533/145630517-f3bd1b70-4b36-4a2d-8e7f-e2c2a756ec9e.png">
I was wondering if `componentDidLoad` is the best place to do the `getSlottedNodes` function.
I know in `va-accordion-item` https://github.com/department-of-veterans-affairs/component-library/blob/8e89df6d262e5ee7cfa2c31355b49cb4f7cb18e0/packages/web-components/src/components/va-accordion/va-accordion-item.tsx#L105 we are using the `onSlotchange` to fire the event which builds the html in the correct format. Wondering if that is the best approach here as well @bkjohnson @andresriveratoro let me know if you think that is the best approach here and if you are also getting that console warn when running the test suite locally

Reply:
I saw that console.warn too and tried using different lifecycle methods to see if I could get rid of it, but it didn't work as intended (nothing displays for children). `componentDidLoad` seems like the correct lifecycle for this use case. I need to be able to get the slot content passed in, but that's not available until after the component has rendered once.

Reply:
Oh, I see. Thanks for that explanation. I'm going to look into it.

Reply:
I've updated it to render breadcrumbs `onSlotchange` instead of `componentDidLoad` for consistency, but I prefer lifecycle methods because they're predictable and seem more correct. I view it similarly to using React's `useRef` vs. something like running `document.getElementById()` inside a lifecycle method. What do you all think? @pdavies88 @k80bowman @bkjohnson

Reply:
I took a look at some other web-components breadcrumbs components just to get an idea of how to resolve this issue. I found [this one](https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/breadcrumbs/breadcrumbs.tsx) from Ionic that looks like it is using Stencil's `componentWillLoad` lifecycle method instead of `componentDidLoad` and `onSlotchange`. Is that something you considered?

Reply:
I tried other lifecycle methods including `componentWillLoad`, and it wouldn't display the breadcrumbs because it couldn't find them. The array length of slotted nodes returned 0.

Reply:
Fun. I, too, prefer the lifecycle methods, but we also don't want extra re-renders. Are there any structural changes you can think of we could make to this component to make it possible to use a lifecycle method? If not, we may have to stick with `onSlotchange`.

Reply:
Using the Ionic component you linked as a reference, I was able to get it working inside `componentWillLoad` by targeting the host element and using `querySelectorAll` for anchor elements instead of targeting the shadow root and the slot (`getSlottedNodes`). The component no longer triggers a re-render. 

`shadowRoot` is null until after the component has loaded. This is why getSlottedNodes was returning an empty array.

Reply:
Does this feel like a better workaround to you?

Reply:
I think this is a great solution for this component and one that we could apply in other components where similar logic is required and when we have a specific element or element type to target.

Reply:
Fantastic! Thank you so much for digging in and figuring it out!